Legacy Executor Checks

revitteth opened the pull request.

hexoscott commented:
Parts of this are still WIP, but good to get it merged as the changes to the stages work fine when running as a sequencer and provide the skeleton for the real verifications to take place

hexoscott committed:
feat(exec): legacy executor check contrived tests

wip: ibs resolve smt retain list

async verification of batches skeleton in place and working

noticed hermez db creation was returning an error when that can't happen any more so update the api there as well

move verification request sending to the verification stage

some tidy around the new verifier stage

fixing some hermez db uses that still expected an error

commenting verifier test temporarily
removing some dead code for now

expect something similar in the future when we tackle witness performance but cleaner without this for now

feat(l1): oldAccInputHash checker (and in hack) (#154)

wire up the verifier with the services it needs

handling a context issue in mapmutation for witness generation and db commit timing

getting oldAccInputHash for executor

sending payload to the executor for verification

making the executor run but ignoring errors for now

also sending full witness whilst we debug the problems with it

ensure we can run without executors in config and just ignore the checks

some tidying for the verification stages

sequencer execute stage log counters consumed and better block sealing

allow some quieter logging when unwinding for witness

nice output of counters for logging

correct logger for verifier

nice output of consumed counters from executor in logs

mandrigin reviewed (changes requested):
small change is to have a command-line flag (true by default) that is "safe" or "strict" mode and requires legacy executors to be there. but user can disable it if needed.

motivation
so in production the defaults always require the executor to be there and we are sure that we aren't producing un-proveable data.
but on dev, we can disable it for our own purposes.

hexoscott committed:
adding strict executor mode flag

mandrigin reviewed (approved):
great, thanks!
